doozer/doozerlib/cli/scan_sources.py

```diff
@@ -3,126 +3,205 @@
-                        rebuild_hint = RebuildHint(RebuildHintCode.BUILD_ROOT_CHANGING, 'Oldest package rpm build was before buildroot change')
-                        runtime.logger.info(f'{dgk} ({eldest_rpm_build}) in {package_name} is older than more recent buildroot change: {build_root_change}')
+                        rebuild_hint = RebuildHint(RebuildHintCode.BUILD_ROOT_CHANGING,
+                                                   'Oldest package rpm build was before buildroot change')
+                        self.runtime.logger.info(f'{dgk} ({eldest_rpm_build}) in {package_name} '
+                                                 f'is older than more recent buildroot change: {build_root_change}')
 
                 if rebuild_hint.rebuild:
-                    add_assessment_reason(meta, rebuild_hint)
-                    changing_rpm_metas.add(meta)
-                    changing_rpm_packages.add(package_name)
+                    self.add_assessment_reason(meta, rebuild_hint)
+                    self.changing_rpm_metas.add(meta)
+                    self.changing_rpm_packages.add(package_name)
 
             elif meta.meta_type == 'image':
                 if rebuild_hint.rebuild:
-                    add_image_meta_change(meta, rebuild_hint)
+                    self.add_image_meta_change(meta, rebuild_hint)
 
             else:
                 raise IOError(f'Unsupported meta type: {meta.meta_type}')
 
-        # To limit the size of the queries we are going to make, find the oldest and newest image
-        oldest_image_event_ts = None
-        newest_image_event_ts = 0
-        for image_meta in runtime.image_metas():
+    def find_oldest_newest(self, koji_api):
+        self.runtime.logger.info('Checking images build time...')
+
+        for image_meta in self.runtime.image_metas():
             info = image_meta.get_latest_build(default=None)
             if info is None:
                 continue
             create_event_ts = koji_api.getEvent(info['creation_event_id'])['ts']
-            if oldest_image_event_ts is None or create_event_ts < oldest_image_event_ts:
-                oldest_image_event_ts = create_event_ts
-            if create_event_ts > newest_image_event_ts:
-                newest_image_event_ts = create_event_ts
+            if self.oldest_image_event_ts is None or create_event_ts < self.oldest_image_event_ts:
+                self.oldest_image_event_ts = create_event_ts
+            if create_event_ts > self.newest_image_event_ts:
+                self.newest_image_event_ts = create_event_ts
 
-            if image_meta in changing_image_metas:
+            if image_meta in self.changing_image_metas:
                 continue  # A rebuild is already requested.
 
-            # Request a rebuild if A is a dependent (operator or child image) of B but the latest build of A is older than B.
+            # Request a rebuild if A is a dependent (operator or child image) of B
```

Comment:
From here, maybe each check can be split into a new method.

Reply:
Done